### DISCUSSION TITLE
Question: which GraphQL Playground to use?

### DISCUSSION THREAD

**Discussion by raarts** [enisdenjo/graphql-ws#274] [Q&A] [answered]
Hi, I am using NestJS, and was following the tutorial there, they 'strongly recommend graphql-ws', but I found that the Apollo Payground (default for nestjs) doesn't support the new protocol yet. Is there an alternative playground that does?

**Comment by enisdenjo** [comment, marked as answer]
GraphQL Playground as of right now does not support `graphql-ws`. There's an issue open (https://github.com/graphql/graphql-playground/issues/1291) and even a PR for supporting both WS subprotocols (`graphql-ws` and `subscriptions-transport-ws`) in parallel (https://github.com/graphql/graphql-playground/pull/1326).

I recommend using GraphiQL instead. [Read here about how to use `graphql-ws` for subscriptions through `@graphiql/toolkit`.](https://github.com/graphql/graphiql/blob/main/packages/graphiql-toolkit/docs/create-fetcher.md#adding-graphql-ws-websockets-subscriptions)

If you want to use _just_ `graphql-ws` for all operations (query, mutation and subscription), [this Gist](https://gist.github.com/enisdenjo/a68312878fdc4df299cb0433c60c1dea) might help you out.
